### PR TITLE
issue/3196 Fixes for branching

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes https://github.com/adaptlearning/adapt_framework/issues/3196

### Fixed
* Check assessment completion on block completion (for presentation components)
* Wait for branching to finish adding children before assessment can be complete

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

File: js/adapt-assessmentArticleModel.js [hunk 1]
```diff
@@ -292,6 +292,9 @@ define([
       questionModels.forEach(questionModel => {
         this._onQuestionCompleted(questionModel, value);
       });
+      if (blockModel.get('_isInteractionComplete')) {
```
Comment (eleanor-heath) [inline js/adapt-assessmentArticleModel.js:295]: could do an early return here

**Review by eleanor-heath — APPROVED** [review]

File: js/adapt-assessmentArticleModel.js [hunk 2]
```diff
@@ -309,6 +311,8 @@ define([
     },
 
     _checkAssessmentComplete: function() {
+      const hasFinalQuestions = (this.get('_requireCompletionOf') !== Number.POSITIVE_INFINITY);
```
Comment (tomgreenfield) [inline js/adapt-assessmentArticleModel.js:314]: `allQuestionsAdded`?

**Review by tomgreenfield — APPROVED** [review]
:eye:

**Pull request merged**